connect-location/src/main/java/com/ifttt/location/ConnectLocation.java

```diff
@@ -0,0 +1,215 @@
+    public void setUp(LocationConfiguration configuration) {
+        if (configuration.connectionApiClient == null) {
+            if (API_CLIENT == null) {
+                ConnectionApiClient.Builder clientBuilder = new ConnectionApiClient.Builder(context);
+                if (configuration.inviteCode != null) {
+                    clientBuilder.setInviteCode(configuration.inviteCode);
+                }
+                if (configuration.inviteCode != null) {
+                    clientBuilder.setInviteCode(configuration.inviteCode);
+                }
+                API_CLIENT = clientBuilder.build();
+            }
+        } else {
+            API_CLIENT = configuration.connectionApiClient;
+        }
+
+        UserTokenAsyncTask task = new UserTokenAsyncTask(configuration.credentialsProvider, () -> {
+            if (configuration.connection != null) {
+                setUpGeofenceForLocationFields(configuration.connection);
+                return;
+            }
+
+            if (configuration.connectionId == null) {
+                throw new IllegalStateException("Connection id cannot be null.");
+            }
+
+            PendingResult<Connection> pendingResult = API_CLIENT.api().showConnection(configuration.connectionId);
+            pendingResult.execute(new PendingResult.ResultCallback<Connection>() {
+                @Override
+                public void onSuccess(Connection result) {
+                    setUpGeofenceForLocationFields(result);
+                }
+
+                @Override
+                public void onFailure(ErrorResponse errorResponse) {
+                    // Handle errors
+                }
+            });
+        });
+        task.execute();
+    }
+
+    private void setUpGeofenceForLocationFields(Connection connection) {
+        List<LocationTriggerField> locationTriggerFields = new ArrayList<>();
+        locationTriggerFields.add( new LocationTriggerField(
+                "id",
+                "location/triggers.enter_or_exit_region_location",
+                    new LocationFieldValue(
+                            37.743520,
+                            -122.431170,
+                            0.0,
+                            "address"
+                    )
+                )
+        );
+
+        // Check if the connection contains location trigger fields before setting up geofences.
+        // Using hard-coded list for testing.
+
+        GeofenceProvider geofenceProvider = new GeofenceProvider(context);
+        geofenceProvider.updateGeofences(locationTriggerFields);
+    }
+
+    public static class LocationConfiguration {
+
+        private final LocationCredentialsProvider credentialsProvider;
+        @Nullable private final ConnectionApiClient connectionApiClient;
+        @Nullable private String connectionId;
+        @Nullable private Connection connection;
+        @Nullable private String inviteCode;
```

Comment:
Not sure if we need inviteCode support either. I think this is in the connect-api level, not the location.